website_sale_cart_expire/models/website.py

```diff
@@ -37,9 +40,12 @@ def _scheduler_website_expire_cart(self):
             website._get_cart_expire_delay_domain() for website in websites
         ]
         carts_to_expire = self.env["sale.order"].search(
-            expression.OR(carts_to_expire_domains)
+            expression.OR(carts_to_expire_domains), order="id ASC",
         )
         # Expire carts
         for cart in carts_to_expire:
-            cart.message_post(body=_("Cart expired"))
-        carts_to_expire.action_cancel()
+            try:
+                cart.action_cancel()
+                cart.message_post(body=_("Cart expired"))
+            except exceptions.except_orm:
+                _logger.exception("Unable to cancel expired cart id %s", cart.id)
```

Comment:
Just ignoring orm exceptions is not good. In case there's one, and in case `action_cancel` performed any write before raising the exception, we'd be commiting that too.

It's solved with a savepoint though:

```suggestion
            try:
                with self.env.cr.savepoint():
                    cart.action_cancel()
                    cart.message_post(body=_("Cart expired"))
            except Exception:
                _logger.exception("Unable to cancel expired cart id %s", cart.id)
```